server/database/fetchDataFromDB.js

```diff
@@ -0,0 +1,69 @@
+// server/database/fetchAllProductInfoFromDB.js
+import DB from './database.js'
+
+const fetchAllProductInfoFromDB = async (page, limit) => {
+  try {
+    const offset = (page - 1) * limit
+    const products = await DB.select('*')
+      .from('products')
+      .whereNotNull('imageUrl')
+      .limit(limit)
+      .offset(offset)
+
+    const productInfoPromises = products.map(async product => {
+      const parts = await DB.select('*')
+        .from('product_partIds')
+        .where('productId', product.productId)
+        .orderBy('id') // Assuming parts are ordered by their ID
+        .limit(1) // Fetching only the first part linked to the product
+
+      if (parts.length > 0) {
+        const part = parts[0]
+        let firstMinQuantity = 0
+        let lastPrice = 0
+        // get the price OBJ from the DB
+        let priceObj = part.pricing
```

Comment:
```suggestion
        const priceObj = part.pricing
```
this should use `const` because you're not changing the value